src/Tests/Receiving/MessagePumpTests.cs

```diff
@@ -61,6 +61,38 @@ public async Task Should_not_process_message_when_lock_expired()
             Assert.That(pumpWasCalled, Is.False);
         }
 
+        [Test]
+        public async Task Should_cancel_receive_token_when_lock_expires_during_processing()
+        {
+            var fakeClient = new FakeServiceBusClient();
+            var fakeReceiver = new FakeReceiver();
+
+            var pump = new MessagePump(fakeClient, new AzureServiceBusTransport(), "receiveAddress",
+                new ReceiveSettings("TestReceiver", new QueueAddress("receiveAddress"), false, false, "error"), (s, exception, arg3) => { }, null);
+
+            bool pumpWasCalled = false;
+
+            await pump.Initialize(new PushRuntimeSettings(1), async (context, token) =>
+                {
+                    pumpWasCalled = true;
+                    await Task.Delay(Timeout.InfiniteTimeSpan, token);
+                },
+                (context, token) => Task.FromResult(ErrorHandleResult.Handled), CancellationToken.None);
+            await pump.StartReceive();
+
+            var receivedMessage = ServiceBusModelFactory.ServiceBusReceivedMessage(messageId: "SomeId", lockedUntil: DateTimeOffset.UtcNow.AddSeconds(60));
+
+            var fakeProcessor = fakeClient.Processors["receiveAddress"];
+            var processingTask = fakeProcessor.ProcessMessage(receivedMessage, fakeReceiver);
+            await fakeProcessor.RaiseMessageLockLost(receivedMessage, new MessageLockLostEventArgs(receivedMessage, new ServiceBusException("Lock Lost", ServiceBusFailureReason.MessageLockLost)));
```

Comment:
Could there be a timing issue between this call and the one above?

Reply:
Even if there is, I am unsure if it is avoidable.

Reply:
Done